crates/re_renderer/src/mesh.rs

```diff
@@ -15,50 +15,55 @@ use crate::{
 };
 
 /// Defines how mesh vertices are built.
-///
-/// Mesh vertices consist of two vertex buffers right now.
-/// One for positions ([`glam::Vec3`]) and one for the rest, called [`mesh_vertices::MeshVertexData`] here
 pub mod mesh_vertices {
     use crate::wgpu_resources::VertexBufferLayout;
     use smallvec::smallvec;
 
-    /// Mesh vertex as used in gpu residing vertex buffers.
-    #[repr(C, packed)]
-    #[derive(Debug, Clone, Copy, bytemuck::Pod, bytemuck::Zeroable)]
-    pub struct MeshVertexData {
-        pub normal: glam::Vec3, // TODO(andreas): Compress. Afaik Octahedral Mapping is the best by far, see https://jcgt.org/published/0003/02/01/
-        pub texcoord: glam::Vec2,
-        // TODO(andreas): More properties? Different kinds of vertices?
-    }
-
     /// Vertex buffer layouts describing how vertex data should be laid out.
     ///
     /// Needs to be kept in sync with `mesh_vertex.wgsl`.
-    pub fn vertex_buffer_layouts() -> [VertexBufferLayout; 2] {
+    pub fn vertex_buffer_layouts() -> [VertexBufferLayout; 4] {
         [
+            // Position:
             VertexBufferLayout {
                 array_stride: std::mem::size_of::<glam::Vec3>() as _,
                 step_mode: wgpu::VertexStepMode::Vertex,
-                attributes: smallvec![
-                    // Position
-                    wgpu::VertexAttribute {
-                        format: wgpu::VertexFormat::Float32x3,
-                        offset: 0,
-                        shader_location: 0,
-                    },
-                ],
+                attributes: smallvec![wgpu::VertexAttribute {
+                    format: wgpu::VertexFormat::Float32x3,
+                    shader_location: 0,
+                    offset: 0,
+                }],
             },
+            // 32 bit color:
             VertexBufferLayout {
-                array_stride: std::mem::size_of::<MeshVertexData>() as _,
+                array_stride: 4,
                 step_mode: wgpu::VertexStepMode::Vertex,
-                attributes: VertexBufferLayout::attributes_from_formats(
-                    1,
-                    [
-                        wgpu::VertexFormat::Float32x3, // Normal
-                        wgpu::VertexFormat::Float32x2, // Texcoord
-                    ]
-                    .into_iter(),
-                ),
+                attributes: smallvec![wgpu::VertexAttribute {
+                    format: wgpu::VertexFormat::Unorm8x4,
+                    shader_location: 1,
+                    offset: 0,
+                }],
```

Comment:
I think it would be nice to have a `VertexBufferLayout::from_single_format` which gets only the format and the `shader_location` passed in. Everything else can be derived from that (assuming `Vertex` step mode ofc)
Up to to you if you prefer that or not.